### ISSUE TITLE
PS2 game always go OPL and not found any game

### ISSUE THREAD

**Issue by saharatch14** [CosmicScale/PSBBN-Definitive-English-Patch#7]
So when install i use HDD 500GB that contain legacy ver(install new ver seem remove i guess?)
[PSBBN-installer.log](https://github.com/user-attachments/files/18149998/PSBBN-installer.log)

then try to add Ps2 & PS1 Games
[game-installer.log](https://github.com/user-attachments/files/18150005/game-installer.log)

also missing art
SLUS_213.59 Metal Gear Solid 3: Subsistence (Disc 1) (Subsistence)

back to PS2 using 39000 JP boot on PSBBN work normal but every time go to play ps2 game always go OPL and no game found but PS1 Game work fine
last time i check in wlaunch seem no file game in each partition (not count OPL stand by)

using Linux Mint 22 Cinnamon on Macbook(2013?) Dual Boot

**Comment by CosmicScale** [comment]
I can't see any issues in the log files. For games to show up in OPL, you need to enable HDD under BDM Devices:

![PXL_20241213_222115962](https://github.com/user-attachments/assets/b19ba6c9-09b4-4433-872b-cecac1ab196e)

Enable this and let me know if the games show in OPL and if you can launch them from there.

**Comment by saharatch14** [comment]
Enable and it didn't not show even try zso format. it mil sec i saw bdm  change GPT/MBR it how it work? maybe i done something wrong when install?

**Comment by CosmicScale** [comment]
Can you mount the OPL exFAT partition on your computer and confirm that there are .zso or .iso files on the drive?

Are you connecting the drive via USB? It could be a block size issue caused by a USB adapter.

**Comment by saharatch14** [comment]
<img width="760" alt="Screenshot_563" src="https://github.com/user-attachments/assets/f04fa583-7970-40ca-912d-6b18238f357d" />

<img width="287" alt="Screenshot_564" src="https://github.com/user-attachments/assets/4612f00e-e0f6-4295-b167-02bc625df794" />

![image](https://github.com/user-attachments/assets/6b87df13-aa40-4500-9858-10c54e25ba37)

yes when game install(on linux) it did show file in OPL and No i didn't use drive via USB

**Comment by Lupus-severus** [comment]
This is the same problem I've been having, using the scph-10350 (Kaico IDE to SATA) and a scph-50000MB NTSCJ console combined with a WD Blue 2TB HDD.

**Comment by CosmicScale** [comment]
I'm really at a loss because it appears the install has gone fine. It seems the problem is with OPL. I use a Kaico IDE to SATA adapter so that definitely isn't the problem.

I can recommend two thing:
1. Connect the drive directly to your PC via the internal SATA connections and try the install again
2. Try a different HDD/SSD

I would highly recommend using an SSD over a HDD, you will not have a smooth experience with a HDD. I use the [Kingston A400 960 GB](https://www.kingston.com/unitedkingdom/en/ssd/a400-solid-state-drive).

**Comment by Lupus-severus** [comment]
I'm definitely leaning towards it being an OPL issue as I've used FHDB before with this hardware config and had no problems, I've also connected and reinstalled directly from internal SATA along with using a usb SATA adapter with no luck.

**Comment by CosmicScale** [comment]
I still think the most likely problem is the sector size. The issue is explained in this video:

[New OPL Update Allows for Any Size Hard Drive, Even 16TB And Higher!](https://youtu.be/nKoRvWKojz0?si=BfxxRIqCUVhWX4VM&t=260)

**Comment by Lupus-severus** [comment]
Ah shit, I ran a lsblk -t and the physical sector size is 4096 with logical being 512. 
That will be the culprit then, although it is strange that it ran with no issue using FHDB and OPL only last week.

I ordered the kingston, if I can get around it then great if not then at least I have a backup.

**Comment by CosmicScale** [comment]
> Ah shit, I ran a lsblk -t and the physical sector size is 4096 with logical being 512. That will be the culprit then, although it is strange that it ran with no issue using FHDB and OPL only last week.
> 
> I ordered the kingston, if I can get around it then great if not then at least I have a backup.

Were you using exFAT in OPL last week?

**Comment by Lupus-severus** [comment]
Just standard HDLBinst, so yeah it's definitely this drive having issues with exFAT

**Comment by saharatch14** [comment]
my issue seem to be HDD can start PSBBN and PS1 Game but can't play PS2 game. It odd than it work well in legacy ver. So i will try SSD like you say
new install 100 and 32 seem get ps2 game to black screen
[PSBBN-installer.log](https://github.com/user-attachments/files/18176633/PSBBN-installer.log)
[game-installer.log](https://github.com/user-attachments/files/18176634/game-installer.log)

![IMG_1796](https://github.com/user-attachments/assets/67fdb3fe-1fba-42aa-b8fc-d7b132594360)

**Comment by CosmicScale** [comment]
> It odd than it work well in legacy ver. 

That's not odd at all. The PS1 games reside on an APA partition, PS2 games reside on an exFAT partition. The problem is definitely the sector size of the exFAT partition not being supported by OPL.

I will add an extra check to the PSBBN Installer script to inform the user if the sector size is incorrect. The only solution is to try a different connection method or try a different drive.

**Comment by CosmicScale** [comment]
I've improved error handling and verification for the PSBBN Installer script in [commit d7b1c98](https://github.com/CosmicScale/PSBBN-Definitive-English-Patch/commit/d7b1c98208f088d8c29f6d4e60e7b495b1796616)

A check for the physical sector size has been added. If it's anything other than 512, a message will now appear with advice on how to proceed. Thanks for the reports. I'll be closing this now.
